tests/unit/configFile.spec.js

```diff
@@ -133,4 +133,239 @@ describe("configFile.js", () => {
+      newRSA.Config.RSAConfigEnable = true;
+
+      // Configuration shouldn't be changed if flag is false.
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      var result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      expect(result.config.RSAConfigEnable).toEqual(
+        defaultSettings.RSA.Config.RSAConfigEnable
+      );
+    });
+
+    it("queries are merged by default and duplicate values aren't added", async () => {
+      var newRSA = _.cloneDeep(defaultSettings.RSA);
+      newRSA.Queries = newRSA.Queries.concat([
+        [-1, "Search Destination IP", "ip.dst=TESTSEARCH", true],
+        [-1, "New Search Destination IP", "new_ip.dst=TESTSEARCH", true],
+      ]);
+
+      // Queries should be merged by default and duplicate values shouldn't be added.
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      var result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      var expectedResult = _.cloneDeep(defaultSettings.RSA);
+      expectedResult.Queries.push([
+        -1,
+        "New Search Destination IP",
+        "new_ip.dst=TESTSEARCH",
+        true,
+      ]);
+      expectedResult = ConfigFile.default.parseQueries(expectedResult.Queries);
+      expect(result.queries).toEqual(expectedResult);
+    });
+
+    it("queries are overridden or ignored depending on corresponding flag", async () => {
+      var newRSA = _.cloneDeep(defaultSettings.RSA);
+      newRSA.Queries = [
+        [-1, "Search Destination IP", "ip.dst=TESTSEARCH", true],
+        [-1, "New Search Destination IP", "new_ip.dst=TESTSEARCH", true],
+      ];
+
+      // Should override queries if flag is override.
+      var settings = (await LocalStore.getOne(StoreKey.SETTINGS)) || {};
+      settings.mergeRSA.queries = "override";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      var result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      var expectedResult = ConfigFile.default.parseQueries([
+        [-1, "Search Destination IP", "ip.dst=TESTSEARCH", true],
+        [-1, "New Search Destination IP", "new_ip.dst=TESTSEARCH", true],
+      ]);
+      expect(result.queries).toEqual(expectedResult);
+
+      // Should ignore queries if flag is ignore.
+      newRSA.Queries = [
+        [-1, "Search Hostname", "alias.host='TESTSEARCH'", true],
+      ];
+      settings.mergeRSA.queries = "ignore";
+      await LocalStore.setOne(StoreKey.SETTINGS, settings);
+      await ConfigFile.default.updateSpecialProvider(
+        StoreKey.RSA_SECURITY,
+        newRSA,
+        "mergeRSA"
+      );
+      result = await LocalStore.getOne(StoreKey.RSA_SECURITY);
+      expect(result.queries).toEqual(expectedResult);
+    });
```

Comment:
Also please split this test case into two: queries are ignored & queries are overridden.